framework/core/js/src/admin/components/AdvancedPage.tsx

```diff
@@ -96,23 +96,49 @@ export default class AdvancedPage<CustomAttrs extends IPageAttrs = IPageAttrs> e
   }
 
   maintenance() {
+    const safeModeExtensionsByConfig =
+      JSON.stringify(app.data.safeModeExtensions) !== JSON.stringify(this.setting('safe_mode_extensions')()) ? app.data.safeModeExtensions : false;
+
     return (
       <FormSection label={app.translator.trans('core.admin.advanced.maintenance.section_label')}>
         <Form>
           {this.buildSettingComponent({
             type: 'select',
             help: app.translator.trans('core.admin.advanced.maintenance.help'),
             setting: 'maintenance_mode',
+            refreshAfterSaving: true,
             options: {
-              0: app.translator.trans('core.admin.advanced.maintenance.options.0'),
-              1: {
+              [MaintenanceMode.NO_MAINTENANCE]: app.translator.trans('core.admin.advanced.maintenance.options.0'),
```

Comment:
Could we use human readable names for the translation keys?

Reply:
Sorry, you probably reviewed a specific PR commit, I did push a third separate commit for that: https://github.com/flarum/framework/pull/3977/commits/7c2a1319ea1cc106f7085e594d2b10b8dd69fb10